core/Model/CalendarModel.vala

```diff
@@ -294,6 +290,36 @@ public class Maya.Model.CalendarModel : Object {
 
     //--- Helper Methods ---//
 
+    /** Set the week_starts_on property: the first day of the week.
+     *
+     * Locale handling is based on information from
+     * https://sourceware.org/glibc/wiki/Locales
+     */
+    private GLib.DateWeekday get_week_start () {
+        // Set the "baseline" for start of week: Sunday or Monday?
+        uint week_day1 = (uint) Posix.NLTime.WEEK_1STDAY.to_string ();
+        var week_1stday = 0; // Default to 0 if unrecognized data
```

Comment:
You could make it clearer by making `week_1stday` an offset between Posix and GLib i.e. `-1` or `0` (and retype/rename appropriately).  Then the formula below becomes e.g.
`var week_start = week_start_posix + offset_posix;`